src/data/set/finite.lean

```diff
@@ -1069,7 +1054,6 @@ lemma finite_range_find_greatest {P : α → ℕ → Prop} [∀ x, decidable_pre
 lemma finite.exists_maximal_wrt [partial_order β] (f : α → β) (s : set α) (h : set.finite s) :
   s.nonempty → ∃ a ∈ s, ∀ a' ∈ s, f a ≤ f a' → f a = f a' :=
 begin
-  classical,
```

Comment:
Has [the bug in `classical`](https://github.com/leanprover-community/mathlib/blob/00dbc7b27bf4a4d2b0990d95e555de0bf23dce8a/src/tactic/core.lean#L1762) been fixed yet @eric-wieser? (Even if it hasn't, this one is still safe to remove since it's the first in the file.)

Reply:
No, I'm no longer sure it's possible to fix.

Reply:
When you use `classical` it permanently adds the classical decidability instances to the current section/namespace.

I think we really ought to revert `classical` to its former behavior ASAP.